### ISSUE TITLE
Using flyToInterpolator with a maxDuration and transitionDuration: auto can crash the page (division by 0)

### ISSUE THREAD

**Issue by RainierXS** [visgl/react-map-gl#1063]
https://github.com/uber/react-map-gl/blob/master/src/utils/transition-manager.js#L233

When transitionDuration is auto, getFlyToDuration from 'viewport-mercator-project' is used to calculate duration which returns 0 when calculated transitionDuration > maxDuration
https://github.com/uber-archive/viewport-mercator-project/blob/master/src/fly-to-viewport.js#L77

using version 5.2.3

**Comment by Pessimistress** [comment]
This was an issue fixed a while ago. What is your viewport-mercator-project version?

**Comment by RainierXS** [comment]
We use the one exported by the package, checking the lock file it is 7.0.1.  So I guess I should have referenced this line instead since 7.0.1 just imports Math.gl/web-mercator (v3.1.3):
https://github.com/uber-web/math.gl/blob/master/modules/web-mercator/src/fly-to-viewport.js#L70

**Comment by Pessimistress** [comment]
@1chandu

**Comment by 1chandu** [comment]
Yeah, this seem to be a bug and fix seem to be simple.

When processing viewport change, we should call interpolator's getDuration instead of just checking mode. At this [line](https://github.com/uber/react-map-gl/blob/6fa913270c7e074ba903e9cfcae51c0572aaf687/src/utils/transition-manager.js#L137)

following code :
```
(transitionDuration > 0 || transitionDuration === 'auto') && Boolean(transitionInterpolator)
```
should be changed to
```
Boolean(transitionInterpolator) && transitionInterpolator.getDuration(...) > 0
```

@RainierXS do you want to give this a shot , I am happy to review the code.

Also not directly related, but I think when mode is `auto` and computed duration happens to be more than `maxDuraiton`, clamping it to `maxDuraiton` seems right option instead of making it `0`.

**Comment by Pessimistress** [comment]
Fixed in 5.2.5
